scipy/signal/tests/test_filter_design.py

```diff
@@ -965,17 +966,21 @@ def test_freqz_sos_design(self):
         w, h = freqz_sos(sos)
         h = np.abs(h)
         w /= np.pi
-        assert_allclose(20 * np.log10(h[w <= 0.1]), 0, atol=3.01)
-        assert_allclose(20 * np.log10(h[w >= 0.6]), 0., atol=3.01)
-        assert_allclose(h[(w >= 0.2) & (w <= 0.5)], 0., atol=1e-3)  # <= -60 dB
+        xp_assert_close(20 * np.log10(h[w <= 0.1]), np.asarray(0.), atol=3.01,
+                        check_shape=False)
+        xp_assert_close(20 * np.log10(h[w >= 0.6]), np.asarray(0.), atol=3.01,
+                        check_shape=False)
+        xp_assert_close(h[(w >= 0.2) & (w <= 0.5)],
+                        np.asarray(0.), atol=1e-3,
+                        check_shape=False)  # <= -60 dB
 
         N, Wn = cheb2ord([0.1, 0.6], [0.2, 0.5], 3, 150)
         sos = cheby2(N, 150, Wn, 'stop', output='sos')
         w, h = freqz_sos(sos)
         dB = 20*np.log10(np.abs(h))
         w /= np.pi
-        assert_allclose(dB[w <= 0.1], 0, atol=3.01)
-        assert_allclose(dB[w >= 0.6], 0., atol=3.01)
+        xp_assert_close(dB[w <= 0.1], 0.0, atol=3.01, check_0d=False, check_shape=False)
+        xp_assert_close(dB[w >= 0.6], 0.0, atol=3.01, check_0d=False, check_shape=False)
```

Comment:
Fair enough to leave out the shape broadcasting, but as mentioned previously, we should use `np.asarray` here instead of `check_0d=False` since this will become `xp.asarray` eventually. The same applies below too.

Reply:
I still do not understand this sentiment. Why is it "better" now that we finally have `check_0d=False`.

Reply:
"better" in the sense that it turns the 0d check on rather than off

Reply:
```suggestion
        xp_assert_close(dB[w <= 0.1], np.asarray(0.0), atol=3.01, check_shape=False)
        xp_assert_close(dB[w >= 0.6], np.asarray(0.0), atol=3.01, check_shape=False)
```

I was just suggesting this. Ultimately it doesn't matter right now, so feel free to leave it like this if you want.